### ISSUE TITLE
Property never used error, on a `private obj: any` parameter of constructor

### ISSUE THREAD

**Issue by iamandrewluca** [Microsoft/TypeScript#13282]
**TypeScript Version:**  2.1.4

**Code**

`tsconfig.json`
```json
{
    "compilerOptions": {
        "module": "none",
        "target": "es5",
        "noImplicitAny": true,
        "strictNullChecks": true,
        "noImplicitThis": true,
        "noImplicitReturns": true,
        "noImplicitUseStrict": false,
        "noEmitOnError": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "removeComments": true,
        "preserveConstEnums": true,
        "sourceMap": false,
        "outFile": "bundle.js"
    }
}
```

`index.ts`
```ts
/// <reference path="main.ts" />
var myObj = {
    test: function() {
        return "666";
    }
}
var test = new Admin.Test(myObj);
```

`main.ts`
```ts
namespace Admin {
    export class Test {
        constructor(private obj: any) {
            console.log(obj.test());
        }
    }
}
```

**Editor:**  Visual Studio Code

**Expected behavior:**
With `noUnusedParameters: true` do not raise an error on used parameter.

**Actual behavior:**
```
main.ts(3,29): error TS6138: Property 'obj' is declared but never used.
```

**What fixes this:** 2 cases
* set `noUnusedLocals` to `false`;
* set `obj` visibility to `public`.

**Comment by iamandrewluca** [comment]
This error is raised not on constructor parameter, but on local `this.obj` that is created
If I add `this.obj.test()` instead of `obj.test()` error dissapears
